File: 2.2. Stack/3. Evaluate Reverse Polish Notation/dory.py
# Runtime 67ms
# Beats  45.92%

# Memory 16.96MB
# Beats 96.96%
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        op = '+-*/'

        for i in tokens:
            if i not in op:
                stack.append(int(i))
            else:
                pop1 = stack.pop()
                pop2 = stack.pop()
                if i == '+':
                    stack.append(pop2 + pop1)
                elif i == '-':
                    stack.append(pop2 - pop1)
                elif i == '*':
                    stack.append(pop2 * pop1)
                elif i == '/':
                    stack.append(int(pop2/pop1))

        return stack[0]

File: 7.1. Intervals/4. Meeting Rooms/dory.cs
# Runtime 132ms
# Beats 61.36%

# Memory 29.0 MB
# Beats 84.09%

/**
 * Definition of Interval:
 * public class Interval {
 *     public int start, end;
 *     public Interval(int start, int end) {
 *         this.start = start;
 *         this.end = end;
 *     }
 * }
 */

public class Solution {
    public bool CanAttendMeetings(List<Driver.Interval> intervals) {
        intervals.Sort((a, b) => a.start.CompareTo(b.start));
        for(int i = 0; i < intervals.Count - 1; i++) {
            if(intervals[i].end > intervals[i+1].start) {
                return false;
            }
        }
        return true;
    }
}
